src/resources/Subscriptions.ts

```diff
@@ -358,6 +367,37 @@ export class Subscriptions extends CRUDResource {
         return this.api.longPolling(promise, successCondition, cancelCondition, callback);
     }
 
+    async pollSubscriptionWithFirstCharge(
+        storeId: string,
+        id: string,
+        data?: SendData<PollParams>,
+        auth?: AuthParams,
+        callback?: ResponseCallback<{ subscription: ResponseSubscription; charge?: ResponseCharge }>
+    ): Promise<{ subscription: ResponseSubscription; charge?: ResponseCharge }> {
+        const subscription = await this.poll(storeId, id, data);
+
+        const charge = await (async () => {
```

Comment:
```js
        const charge = !hasImmediateCharge(subscription)
            ? null
            : await this.api.longPolling(
                () => this.charges(storeId, id),
                (charges) => !!charges.items.length
            ).then(({ items: charges }) => this.api.longPolling(
                () => this.chargesResource.get(charges[0].storeId, charges[0].id, data, auth),
                ({ status }) => status !== ChargeStatus.PENDING
            ));
```

This is more compact than the other version and avoids the immediately called anonymous function, but both are ultimately equivalent!

Reply:
Sure, let me refactor this part 🙂 Thanks!

Reply:
Also used `charges.poll` instead of re-implementing the long polling for this resource. Thanks for pointing this function out 🙂